### PR TITLE
Color percentage indicator

### PR DESCRIPTION
**Pull request by MarcoGorelli**
Fixes https://github.com/pytest-dev/pytest/issues/6097.

### CODE REVIEW

**Comment by MarcoGorelli** [comment]
I'm aware I'm supposed to write tests before writing code, I just haven't (yet) figured out how to test this, sorry

**Comment by blueyed** [comment]
No worries, if you're lucky you can adjust an existing one.
(I've edited the original comment, to add a reference, and remove the boilerplate)
Thanks for working on this! :)

**Comment by blueyed** [comment]
While working on this we should keep in mind to use it also for the "SKIPPED", "FAILED", "XFAIL", etc prefixes in the short test summary.

**Comment by blueyed** [comment]
CI passes (except for the flaky hypothesis issue), so you need a new test.
I suggest using one based on `monkeypatch.setenv("PY_COLORS", "1")`, so that color output is forced.

**Comment by MarcoGorelli** [comment]
Thanks, I was wondering about the hypothesis issue.

OK, I'll give that a go and will let you know

File: testing/test_terminal.py [hunk 1]
```diff
@@ -1835,13 +1835,13 @@ def test_collecterror(testdir):
...
             "*= ERRORS =*",
             "*_ ERROR collecting test_collecterror.py _*",
             "E   SyntaxError: *",
             "*= short test summary info =*",
             "ERROR test_collecterror.py",
-            "*! Interrupted: 1 errors during collection !*",
+            "*! Interrupted: 1 error during collection !*",
             "*= 1 error in *",
         ]
     )
```
Comment (blueyed) [inline testing/test_terminal.py]: Yes, bad/outdated merge.
Cherry-picked into features in 9303de8.

**Comment by MarcoGorelli** [comment]
I've added tests with `monkeypatch.setenv("PY_COLORS", "1")`, is this what you had in mind?

**Comment by blueyed** [comment]
> I've added tests with `monkeypatch.setenv("PY_COLORS", "1")`, is this what you had in mind?

Yes.  Although I think we should them more readable (via format strings for color escapes), so that it would use `"{red}"` there then.
Also I am not sure if the base/default test is really needed, and if it would be faster to have them all in a single run (have not looked at what the "many files" fixture really does).
Looking forward to testing this out later..!

**Comment by MarcoGorelli** [comment]
Ok, I've removed two of the tests and have kept the one that has all three colors in it (green, red, yellow). I've also added constants to make the tests more readable.

Just realised that pytest-CI checks for Python 3.5 to work too, so no f-strings....I'll change this later today

**Comment by blueyed** [comment]
Nice!  You also need a changelog then (see `changelog/README.rst`).

**Comment by MarcoGorelli** [comment]
Thanks! Changelog added

**Comment by blueyed** [comment]
Thanks, will review it properly later - you could however squash and force-push it already (sorry, I'd like we could just do it here (#4361)).

**Comment by blueyed** [comment]
Awesome!

**Pull request merged**